data/campaigns/Eastern_Invasion/utils/traits.cfg

```diff
@@ -0,0 +1,312 @@
+                    [/filter]
+
+                    [object]
+                        id=mercenary_buff
+                        [effect]
+                            apply_to,increase_damage=attack,+30%
+                        [/effect]
+                    [/object]
+                [/modify_unit]
+            [/then]
+            [else]
+                [modify_unit]
+                    [filter]
+                        side,trait=1,mercenary
+                    [/filter]
+
+                    [object]
+                        id=mercenary_debuff
+                        [effect]
+                            apply_to,increase_damage=attack,-30%
+                        [/effect]
+                    [/object]
+                [/modify_unit]
+            [/else]
+        [/if]
+    [/event]
+#enddef
+
+#define RESIST_DAMAGE DAMAGE PERCENTAGE
+    [resistance]
+        affect_self=yes
+        [filter_self]
+            [filter_adjacent]
+                # ok, ok, yes, I know, this will trigger even if you're attacking non-undead as long as you're adjacent to undead.
+                # but that's such an unusual edge case, and I don't know how to filter this by enemy
+                trait=undead
+
+                [or]
+                    type_adv_tree=Dark Adept,Vampire Bat
+                [/or]
+            [/filter_adjacent]
+        [/filter_self]
+        [filter_second_weapon]
+            damage={DAMAGE}
+        [/filter_second_weapon]
+        add={PERCENTAGE}
+        max_value=100 # required by WML
+    [/resistance]
+#enddef
+#define TRAIT_SURVIVOR
+    [trait]
+        id=survivor
+        male_name= _ "survivor"
+        female_name= _ "female^survivor"
+        description= _ "<span color='#00FF00'>+1</span> damage to and <span color='#00FF00'>-1</span> damage from undead, necromancers, and bats.
+
+Against all odds, this unit has endured the terrors and tortures of Mal-Ravanal’s prisons and lived to tell the tale. Survivors bear a heavy burden from their time in captivity, but have also learned much of the strengths and weaknesses of their former captors."
+
+        # nice and simple, just give +1 damage vs undead
+        [effect]
+            apply_to=attack # all attacks
+            [set_specials]
+                mode=append
+                [damage]
+                    apply_to=self
+                    [filter_opponent] # note that I also use this filter in the implementation for the Sentinel shield
+                        trait=undead
+
+                        [or]
+                            type_adv_tree=Dark Adept,Vampire Bat
+                        [/or]
+                    [/filter_opponent]
+                    add=1
+                [/damage]
+            [/set_specials]
+        [/effect]
+
+        # resistance only works by percentages, so to implement -1 damage
+        # I need a separate hard-coded percentage for each possible attack
+        # TODO: make this less janky
```

Comment:
Yes please…

I don't know exactly what `RESIST_DAMAGE` is as of the time of writing, but I'm guessing this can be solved with some WFL?